src/config.rs

```diff
@@ -1,33 +1,140 @@
-use std::path::{Path, PathBuf};
+use serde::{Deserialize, Serialize};
+use std::{
+    collections::HashMap,
+    fs::File,
+    path::{Path, PathBuf},
+    process,
+};
+
+/// Restart conditions for a service.
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "lowercase")]
+enum Restart {
+    Never,
+    Always,
+    Unexpected,
+}
 
-#[derive(Debug)]
 /// Configuration structure used to run a task.
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(deny_unknown_fields)]
+#[serde(default)]
 pub struct Config {
     /// Command to run.
     cmd: String,
     /// Number of processors this task should run with.
     // u16 is fine because we are not expecting a machine to run more
     // than 2^16 proc at any single time.
     numprocs: u16,
+    /// Program mask running under
+    umask: i32,
```

Comment:
Should we be using using `u16` (unsigned and limit to 16 because 0000 < u16 < 7777) ?

Reply:
Yeah sure, could do.
The best would have been to use the `umask` crate and use the format they mention in the doc [https://crates.io/crates/umask](https://crates.io/crates/umask)
We could have wrote "rw-r--r--" . Awesome, unfortunately they do not implement some `serde` traits. I do not have sufficient knowledge to implement trait on existing types. If we could do this, it would be awesome. Do you have any idea how to achieve this ?

Reply:
You can't implement a Trait on a type that is not declared in your module. You can create a `Wrapper` and implement the trait for that `Wrapper` (this is a workaround commonly used).
However, I think we need to use the minimum amount of external crates (as mentioned by the subject). I think working with numbers is fine (though if you really want to, we could create a simple function to translate from `rwx` to numbers, but I don't think that's necessary).
What do you think?

Reply:
Sure !

Reply:
Btw I think my comment about the Trait thing is wrong... I need to check again, but I think it's a bit more complicated than that :p